File: backend/core/src/modules/ai-usage/ai-usage.controller.ts
import { Controller, UseGuards } from '@nestjs/common';
import { UserScopedCrudController } from '../../common/access/user-scoped-crud.controller';
import { AiUsageService } from './ai-usage.service';
import { AiUsage } from '../../entities';
import { JwtAuthGuard } from '../auth/jwt-auth.guard';

@UseGuards(JwtAuthGuard)
@Controller('ai-usage')
export class AiUsageController extends UserScopedCrudController<AiUsage> {
  constructor(service: AiUsageService) {
    super(service);
  }
}

File: backend/core/src/main/java/store/pjcloud/cowrite/core/controller/AiUsageController.java
package store.pjcloud.cowrite.core.controller;

import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import store.pjcloud.cowrite.core.common.UserScopedCrudController;
import store.pjcloud.cowrite.core.entity.AiUsage;
import store.pjcloud.cowrite.core.service.AiUsageService;

@RestController
@RequestMapping("/ai-usage")
public class AiUsageController extends UserScopedCrudController<AiUsage> {
    public AiUsageController(AiUsageService service) {
        super(service);
    }
}
